src/batou/environment.py

```diff
@@ -566,6 +566,17 @@ def configure(self):
                 UnusedResources.from_context(self.resources.unused)
             )
 
+        # if any of Component.instances has ._prepared == False, then
```

Comment:
I would actually make this an error in the configuration phase as this is against the consistency we want to guarantee.